### PR TITLE
TMDM-13991 Advanced Search does not work with AND query on Foreign Key Field

### PR DESCRIPTION
**Pull request by hw-zhu**
https://jira.talendforge.org/browse/TMDM-13991
**What is the current behavior?** (You should also link to an open issue here)
Lucene query need to use full path as condition based on above description,but now only have the Node and the element below the Node when the current FieldMetadata element to be parsed

**What is the new behavior?**
Modify related method, get the current field's containingType, if it's not empty, continue to rise up until it's containingType is empty, finally, get the absolute path of the field, get the absolute path for the current fieldName.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java [hunk 1]
```diff
@@ -415,6 +415,38 @@ public Void visit(EnumerationFieldMetadata enumField) {
...
+     * Person
+     *      |__Stores
+     *              |__Store
+     *                      |__Name
+     *      |__Age                
+     * </pre>
+     * @return: Person.Stores.Store.Name
+     */
+    private String getFullPathName(FieldFullText fieldFullText) {
+        FieldMetadata tempFieldMetadata = fieldFullText.getField().getFieldMetadata();
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java]: How about use variable name `filedMetadata`?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java [hunk 2]
```diff
@@ -415,6 +415,38 @@ public Void visit(EnumerationFieldMetadata enumField) {
...
+        StringBuilder fullPathFieldName = new StringBuilder();
+        while (tempFieldMetadata instanceof ReferenceFieldMetadata) {
+            String tempFieldName = tempFieldMetadata.getName();
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java]: How about use variable as `fieldName`?

Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java]: done

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java (hunk 1 above)
Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/LuceneQueryGenerator.java]: done

**Pull request merged**